### PR TITLE
Save user manual connections to be scanned 

### LINKED ISSUE #818: Remember Manual Connections Across App Restarts
## Summary

As I understand, Ping-Viewer only looks at 192.168.2.2 for connecting to PingProxy automatically. If the device running PingProxy has a different address, you have to enter it manually every time Ping-Viewer is started. 

It would be nice if there were the ability to remember manually entered configurations across restarts of the app. Perhaps only save them if we receive a valid Ping packet in return. Alternatively, there could be a button in the settings panel to save a connection as a 'favorite'.

Ideally, these saved connections or *favorites* would be scanned when the app is restarted so that the online devices populate in the device manager, but devices that are off/not responding don't show up. 

## Additional information

There are lots of situations where Ping devices may be on various different hosts, not just a standard BlueROV2. For example, if you're connected to an ROV and a surface vessel at the same time. Or multiple ROVs. Or maybe you have 10 buoys on a lake and each of them has a Pi and Ping.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #818 
out

### CODE REVIEW

File: src/sensor/ping.cpp [hunk 1]
```diff
@@ -103,8 +103,6 @@ void Ping::startPreConfigurationProcess()
 
     // Start periodic request timer
     _periodicRequestTimer.start();
-
-    // Save configuration
```
Comment (jaxxzer) [inline src/sensor/ping.cpp]: squash with parent?

File: src/settings/settingsmanager.cpp [hunk 2]
```diff
@@ -17,10 +18,45 @@ SettingsManager::SettingsManager()
...
+            }),
+        _lastLinkConfigurations.end());
+
+    // Move to QVariantList before saving
+    // This is necessary since QList<QDataStream operator> does not work as expected
+    QVariantList variants;
+    for (const auto linkConfiguration : _lastLinkConfigurations) {
+        variants.append(QVariant::fromValue(linkConfiguration));
+    }
+    set("lastLinkConfigurations", variants);
```
Comment (jaxxzer) [inline src/settings/settingsmanager.cpp]: why use `set` instead of `=`?

File: src/settings/settingsmanager.cpp [hunk 3]
```diff
@@ -17,10 +18,45 @@ SettingsManager::SettingsManager()
...
+    for (const auto variant : variantLastLinkConfigurations) {
+        _lastLinkConfigurations.append(variant.value<LinkConfiguration>());
+    }
+}
+
+void SettingsManager::saveLinkConfigurations()
+{
+    // Remove duplicated before saving it
+    _lastLinkConfigurations.erase(
+        std::unique(_lastLinkConfigurations.begin(), _lastLinkConfigurations.end(),
```
Comment (jaxxzer) [inline src/settings/settingsmanager.cpp]: nice

File: src/settings/settingsmanager.cpp (hunk 2 above)
Comment (patrickelectric) [reply src/settings/settingsmanager.cpp]: `Set` check if the variable exist or not, print a debug message related to the variable existence, the old and new value of such variable and also forces the sync of the settings.

**Pull request merged**